substrate/client/rpc-spec-v2/src/chain_head/subscription/inner.rs

```diff
@@ -750,22 +750,46 @@ impl<Block: BlockT, BE: Backend<Block>> SubscriptionsInner<Block, BE> {
 		}
 	}
 
-	pub fn unpin_block(
+	pub fn unpin_blocks(
 		&mut self,
 		sub_id: &str,
-		hash: Block::Hash,
+		hashes: Vec<Block::Hash>,
 	) -> Result<(), SubscriptionManagementError> {
-		let Some(sub) = self.subs.get_mut(sub_id) else {
+		// Check if the subscription ID is valid or not.
+		if self.subs.get(sub_id).is_none() {
```

Comment:
hmm, how expensive is it to clone the SubscriptionState here? 
It's a bit annoying to do that many HashMap looks here

Reply:
To work around this, since we cannot use `RefCell` because `ChainHead` needs to be `Send`, I've removed the mappings from the subscription first, which requires only one `HashMap::get`.

Then, I've removed the global mappings of the hashes. The downside if that we need to do `O(3*n)` instead of ideal `O(2*n)`, but given the hashes provided in unpin are bounded to the total of the subscription I would say this is pretty good in terms of performance

Reply:
Ok, yeah realized that now but as you said the number pinned blocks are bounded and should fast enough I reckon as well.